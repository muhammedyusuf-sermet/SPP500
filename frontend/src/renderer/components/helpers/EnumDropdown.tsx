import * as React from 'react';

import 'bulma/css/bulma.css';

import { Input, Select } from '@material-ui/core';

export interface IEnumDropdownProps {
	disabled?: boolean,
	name: string,
	selected?: string,
	options: string[]
	onChange: (name: string, selectOption?: string) => void
}

export interface IEnumDropdownState {
	selected: string
}

export class EnumDropdown extends React.Component<IEnumDropdownProps, IEnumDropdownState> {
	constructor(props: IEnumDropdownProps) {
		super(props);
		this.state = {
			selected: props.selected ? props.selected : 'Default'
		};
	}

	componentWillReceiveProps(nextProps: IEnumDropdownProps) {
		if (this.state.selected != nextProps.selected)
			this.setState({
				selected: nextProps.selected ? nextProps.selected : 'Default'
			});
	}

	handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
		this.setState({selected: event.target.value});
		this.props.onChange(this.props.name, event.target.value);
	}

	render() {
		return (
			<Select
				native={true}
				id={this.props.name}
				disabled={this.props.disabled}
				onChange={this.handleChange}
				value={this.state.selected}
				input={<Input name={this.props.name} id={this.props.name} />} >
				<option
					id={this.props.name+'.'+'Default'}
					value='' >
					Default
				</option>
				{this.props.options.map(option =>
					<option
						key={this.props.name+'.'+option}
						id={this.props.name+'.'+option}
						value={option}>
						{option.replace(/([A-Z])/g, ' $1').trim()}
					</option>
				)}
			</Select>
		);
	}
}