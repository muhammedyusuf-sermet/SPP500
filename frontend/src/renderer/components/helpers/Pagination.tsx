import * as React from 'react';
import '../../css/platform/pages/catalogue-pagination.css';


export interface IPaginationState {
	page: number,
}

export class Pagination extends React.Component<any, IPaginationState> {
	constructor(props: any) {
		super(props);
		this.state = {
			page: 0,
		}
	}

	getTotalPages() {
		return this.props.getTotalPages();
	}

	previousPage() {
		if(this.state.page > 0){
			let newPage = this.state.page-1;
			this.setState({ page: newPage});
			this.props.onPageChange(newPage);
		}
	}

	nextPage() {
		let totalPages = this.getTotalPages();
		if(this.state.page < totalPages){
			let newPage = this.state.page+1;
			this.setState({ page: newPage});
			this.props.onPageChange(newPage);
		}
	}

	render() {
			return (
				<div id="paginated-catalogue-navigation">
					<h3>Page No: {this.state.page+1}</h3>
					<a onClick={() => this.previousPage()} id="previousPageButton" className="previous">&laquo; Previous</a>
					<a onClick={() => this.nextPage()} id="nextPageButton" className="next">Next &raquo;</a>
				</div>
			)
	}
}
