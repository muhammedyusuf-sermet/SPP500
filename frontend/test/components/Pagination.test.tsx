import * as React from "react"
import { shallow, ShallowWrapper } from 'enzyme';
import {Pagination, IPaginationState} from '../../src/renderer/components/helpers/Pagination';

describe('Test the Pagination Component', () => {
	let paginationInstance: ShallowWrapper<any, IPaginationState, Pagination>;

	beforeEach(() => {
		paginationInstance = shallow(<Pagination/>);
		paginationInstance.setProps({ getTotalPages: () => {return 5;}, onPageChange: () => {return null;} })
	})

	it('renders without crashing', () => {
		expect(paginationInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(paginationInstance).toMatchSnapshot();
	});

	it('should update the state variables when Next button is clicked', () => {
		paginationInstance.instance().setState({ page: 2});
		paginationInstance.find('a#nextPageButton').simulate('click', {preventDefault: () => {}});

		expect(paginationInstance.state('page')).toEqual(3);
	});

	it('should update the state variables when Previous button is clicked', () => {
		paginationInstance.instance().setState({ page: 2 });
		paginationInstance.find('a#previousPageButton').simulate('click', {preventDefault: () => {}});

		expect(paginationInstance.state('page')).toEqual(1);
	});
});
