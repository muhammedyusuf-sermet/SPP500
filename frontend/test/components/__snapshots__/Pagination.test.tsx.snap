// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Pagination Component renders correctly when the page is loaded 1`] = `
<div
  id="paginated-catalogue-navigation"
>
  <h3>
    Page No: 
    1
  </h3>
  <a
    className="previous"
    id="previousPageButton"
    onClick={[Function]}
  >
    « Previous
  </a>
  <a
    className="next"
    id="nextPageButton"
    onClick={[Function]}
  >
    Next »
  </a>
</div>
`;
